### PR TITLE
cmds/mount: change usage/error reporting

### PR DESCRIPTION
**Pull request by fozog**
changed usage string to reflect exact rules
inform user if filesystem type not supported

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,6 +82,13 @@ qemu-system-x86_64 -kernel path/to/kernel -initrd /tmp/initramfs.linux_amd64.cpi
 Note that you do not have to build a special kernel on your own, it is
 sufficient to use an existing one. Usually you can find one in `/boot`.
 
+>NOTE: you can compress the initramfs but for xz compression, the kernel has some restrictions
```
Comment (rminnich) [inline README.md]: can you take this part out and put it in a different commit with a message? It's best to limit commits to a single subject. Thanks.

File: cmds/mount/mount.go [hunk 2]
```diff
@@ -25,14 +28,38 @@ var (
...
+			fs = tokens.Text()
+		}
+		// just check the last token of the line
+		if fs == originFS {
+			return originErr
+		}
+	}
+	return errors.New("Unknown filesystem, check /proc/filesystems for supported ones")
+}
+
```
Comment (rminnich) [inline cmds/mount/mount.go]: See https://play.golang.org/p/ZPC0chiVOGV for a slightly less involved way to do this.
You can start with 
b, err := ioutil.ReadFile("/proc/filesystems")
if err != nil {
return
}
for s := range strings.Split(string(s), "\n")) {
...
}

Up to you, this code seems like  more work than you need. In any event, see the comment below.

File: cmds/mount/mount.go [hunk 3]
```diff
@@ -48,7 +75,11 @@ func main() {
 	if *ro {
 		flags |= unix.MS_RDONLY
 	}
-	if err := unix.Mount(a[0], a[1], *fsType, flags, strings.Join(data, ",")); err != nil {
+	if *fsType == "" {
+		log.Printf("No file system type provided!\n")
+		log.Fatalf("Usage: mount [-r] [-o mount options] -t fstype dev path")
+	} else if err := unix.Mount(a[0], a[1], *fsType, flags, strings.Join(data, ",")); err != nil {
```
Comment (rminnich) [inline cmds/mount/mount.go]: The problem here is that there can be lots of reasons for an error on mount, and this is going to confuse people if they're told that it's a nonexistent file system if that's not the real problem. 

And since you have that nice function, why not use it BEFORE the call to mount so you can tell them right away that it's an unknown file system? 

Since log.Fatalf exits you don't need an else.

Oh and ... one reason -t may not be mandatory in fiuture is because of bind mounts, which this command does not support but which also don't require a file system name. So -t at some point may go back to being optional. Mount is such fun :-)

File: README.md (hunk 1 above)
Comment (fozog) [reply README.md]: That was my intent. I am new to github PR process. So I naively thought creating a second pull request would be targeted at the last commit only... will do.

File: cmds/mount/mount.go (hunk 3 above)
Comment (fozog) [reply cmds/mount/mount.go]: The code you point to is verification that a -t has been supplied by the user. so that is a check before.

I shall do things differently in the part that actually document the error.
My business space is embedded so boot time is important (you don't want to wait for your rear camera when you start your car and want to backup).
Even this is little work, I don't want checks to slow down regular work programmed in script because I want to help users.
So, as a "philosophy" I prefer to do "post-portem" documentation rather than screening things at the beginning. Except if checks are here to prevent harm of the command to be executed.

If you agree, I'll propose a different way to document the error after it happens.

**Comment by hugelgupf** [comment]
Would you mind rebasing and dropping the merge commits?

**Comment by fozog** [comment]
I have split this commit in two : #680 and #681
